geopandas/sindex.py

```diff
@@ -243,6 +244,28 @@ def query_bulk(self, geometry, predicate=None, sort=False):
                 input_geometry_index.extend([i] * len(res))
             return np.vstack([input_geometry_index, tree_index])
 
+        def intersection(self, coordinates, objects=False):
+            """Find tree geometries that intersect the input coordinates.
+
+            Parameters
+            ----------
+            coordinates : sequence or array
+                Sequence of the form (min_x, min_y, max_x, max_y)
+                to query a rectangle or (x, y) to query a point.
+            objects : True or False
+                If True, return the label based indexes. If False, integer indexes
+                are returned.
+            """
+            if objects:
+                warn(
+                    "A future version of GeoPandas will deprecate the `objects`"
```

Comment:
How about

```
'objects' is deprecated and will be replaced by a different interface to labels in the sindex geometries in an upcoming release.
```

I don't know that the "If you use this option" is necessary here; this doesn't get added to other deprecation notices.

Reply:
My thoughts were that we do not really know who uses this interface and why. I thought it would be good to get some issue reports that might help shape what that future implementation of `loc/label` based indexing will look like, or if it is needed at all.

Reply:
I think we can just say it will be removed in the future. But it might be good to indicate what to do instead (how to achieve the same behaviour without using the keyword), so it's clear for users how to replace their usage of it.

Reply:
So something like:
```
`objects` is deprecated and will be removed in a future version. Use `iloc` to index your GeoSeries/GeoDataFrame using integer indexes returned by `intersection`.
```
?